File: Cloe y la Secuencia/Cloe y la Secuencia.pas
program Hd;

function bpow(base, exp: Int64): Int64;
var
    result: Int64;
begin
    result := 1;
    while exp > 0 do
    begin
        if exp mod 2 = 1 then
        begin
            result := result * base;
        end;
        base := base * base;
        exp := exp div 2;
    end;
    bpow := result;
end;

var
    aux, k, n, m: Int64;
begin
    readln(n, k);
    m := bpow(2, n - 1);
    while m <> k do
    begin
        if k > m then
        begin
            aux := k - m;
            k := m - aux;
        end;
        n := n - 1;
        m := bpow(2, n - 1);
    end;
    writeln(n);
end.

File: Cloe y la Secuencia/Cloe y la Secuencia.c
#include<stdio.h>
#define Hd main()

long long int bpow(long long int base, long long int exp) {
    long long int result = 1;
    while (exp > 0) {
        if (exp % 2 == 1) {
            result *= base;
        }
        base *= base;
        exp /= 2;
    }
    return result;
}

Hd
{
    long long int aux,k,n,m;
    scanf("%lld%lld",&n,&k);
    m=bpow(2,n-1);
    while(m!=k)
    {
        if(k>m)
        {
            aux=k-m;
            k=m-aux;
        }
        n--;
        m=bpow(2,n-1);
    }
    printf("%lld",n);
}